src/docfx/lib/path/FilePath.cs

```diff
@@ -48,11 +48,25 @@ public FilePath(string path, string commit, FileOrigin origin)
 
         public FilePath(string path, string dependencyName)
         {
-            Path = PathUtility.NormalizeFile(path);
+            Path = PathUtility.NormalizeFile(System.IO.Path.Combine(dependencyName, path));
             DependencyName = dependencyName;
             Origin = FileOrigin.Dependency;
         }
 
+        /// <summary>
+        /// Gets the path relative to docset root or dependency docset root
+        /// </summary>
+        public string GetPathToOrigin()
```

Comment:
what about a property? e.g.
```code
public string PathToOrigin => 
         Origin == FileOrigin.Dependency 
         ? PathUtility.NormalizeFile(System.IO.Path.GetRelativePath(DependencyName, Path))
         : Path;
```

Reply:
I took Yufei's suggestion to change it to a method, avoid Serialization for unused properties.